frontend/src/shared/video-card-image.tsx

```diff
@@ -0,0 +1,25 @@
+"use client";
+
+interface Props {
+  videoId: string;
+  progress?: number;
+}
+
+/**
+ * Renders the thumbnail of a video or the progress of the video if it is being processed
+ */
+export default function VideoCardImage({ progress, videoId }: Props) {
+  
+  if (!progress) {
+    return <img src={`/api/clips/${videoId}/thumbnail.jpg`} />
+  }
+
+  if (progress === -1) {
+    return <div style={{ userSelect: "none" }}>Queued</div>
+  }
+
+  // @TODO: Since we have a thumbnail here we can maybe add some opacity to it and show the progress on top of it
```

Comment:
We should make issues from TODO's and potentially even link to issue here in the TODO, that could make them a little large though.

Regardless, we should create issues so we don't lose them "in the code" in the form of TODO's